src/template.c

```diff
@@ -19,19 +28,26 @@ std::vector<std::string> template_dfa(CharacterVector x) {
     code_percentOp,
     code_comment,
     code_comment_oneCloseBracket
-  };
+  } State;
 
-  if (x.length() != 1) {
-    stop("Input HTML must be a character vector of length 1");
+  if (Rf_xlength(x_sxp) != 1) {
+    Rf_error("Input HTML must be a character vector of length 1");
   }
-  std::string input = Rcpp::as<std::string>(x[0]);
-  std::vector<std::string> pieces(0);
+
+  SEXP str;
+  SEXP pieces = Rf_allocVector(STRSXP, 10);
```

Comment:
A used a initial allocation of 10, if you think that is too high we can lower it.

Reply:
10 seems reasonable to me.  Does this need a `PROTECT`?

Reply:
No, we protect it immediately below.